pkg/installer/daemonset.go

```diff
@@ -79,33 +82,32 @@ func createOrUpdateNodeAPIServerSecrets(ctx context.Context, args *Args) error {
 	)
 }
 
-func doCreateDaemonset(ctx context.Context, args *Args) error {
-	if err := createOrUpdateNodeAPIServerSecrets(ctx, args); err != nil {
-		return err
-	}
-
+func newSecurityContext(seccompProfile string) *corev1.SecurityContext {
 	privileged := true
 	securityContext := &corev1.SecurityContext{Privileged: &privileged}
-
-	if args.SeccompProfile != "" {
+	if seccompProfile != "" {
 		securityContext.SeccompProfile = &corev1.SeccompProfile{
 			Type:             corev1.SeccompProfileTypeLocalhost,
-			LocalhostProfile: &args.SeccompProfile,
+			LocalhostProfile: &seccompProfile,
 		}
 	}
 
-	volumes := []corev1.Volume{
-		newHostPathVolume(volumeNameSocketDir, newPluginsSocketDir(kubeletDirPath, consts.Identity)),
+	return securityContext
+}
+
+func getVolumesAndMounts(pluginSocketDir, certsSecretName string) (volumes []corev1.Volume, volumeMounts []corev1.VolumeMount) {
+	volumes = []corev1.Volume{
+		newHostPathVolume(volumeNameSocketDir, pluginSocketDir),
 		newHostPathVolume(volumeNameMountpointDir, kubeletDirPath+"/pods"),
 		newHostPathVolume(volumeNameRegistrationDir, kubeletDirPath+"/plugins_registry"),
 		newHostPathVolume(volumeNamePluginDir, kubeletDirPath+"/plugins"),
 		newHostPathVolume(volumeNameAppRootDir, appRootDir),
 		newHostPathVolume(volumeNameSysDir, volumePathSysDir),
 		newHostPathVolume(volumeNameDevDir, volumePathDevDir),
 		newHostPathVolume(volumeNameRunUdevData, volumePathRunUdevData),
```

Comment:
we also need to mount `/var/lib/direct-csi/` in legacy mode. Else, the drive controller cannot `remove` the drive (kubectl directpv remove). ie, it cannot umount the primary mount `/var/lib/direct-csi/` as the path is not mounted to the container.

The following error will be seen while umounting,

```log
E1215 07:07:59.096571  592613 listener.go:181] no such file or directory
```

It worked after applying the following diffs

```diff
diff --git a/pkg/consts/consts.go b/pkg/consts/consts.go
index 946263c1..01e19d93 100644
--- a/pkg/consts/consts.go
+++ b/pkg/consts/consts.go
@@ -64,6 +64,9 @@ const (
        // AppRootDir is application root directory.
        AppRootDir = "/var/lib/" + AppName
 
+       // LegacyAppRootDir denotes the legacy root directory.
+       LegacyAppRootDir = "/var/lib/direct-csi"
+
        // UdevDataDir is Udev data directory.
        UdevDataDir = "/run/udev/data"
 
diff --git a/pkg/consts/consts.go.in b/pkg/consts/consts.go.in
index b424646c..7320535a 100644
--- a/pkg/consts/consts.go.in
+++ b/pkg/consts/consts.go.in
@@ -62,6 +62,9 @@ const (
        // AppRootDir is application root directory.
        AppRootDir = "/var/lib/" + AppName
 
+       // LegacyAppRootDir denotes the legacy root directory.
+       LegacyAppRootDir = "/var/lib/direct-csi"
+
        // UdevDataDir is Udev data directory.
        UdevDataDir = "/run/udev/data"
```

```diff
diff --git a/pkg/installer/daemonset.go b/pkg/installer/daemonset.go
index a46dd398..ba9a947f 100644
--- a/pkg/installer/daemonset.go
+++ b/pkg/installer/daemonset.go
@@ -40,7 +40,9 @@ const (
        volumeNameRegistrationDir          = "registration-dir"
        volumeNamePluginDir                = "plugins-dir"
        volumeNameAppRootDir               = consts.AppName + "-common-root"
+       volumeNameLegacyAppRootDir         = "direct-csi-common-root"
        appRootDir                         = consts.AppRootDir + "/"
+       legacyAppRootDir                   = consts.LegacyAppRootDir + "/"
        volumeNameSysDir                   = "sysfs"
        volumeNameDevDir                   = "devfs"
        volumePathDevDir                   = "/dev"
@@ -102,6 +104,7 @@ func getVolumesAndMounts(pluginSocketDir, certsSecretName string) (volumes []cor
                newHostPathVolume(volumeNameRegistrationDir, kubeletDirPath+"/plugins_registry"),
                newHostPathVolume(volumeNamePluginDir, kubeletDirPath+"/plugins"),
                newHostPathVolume(volumeNameAppRootDir, appRootDir),
+               newHostPathVolume(volumeNameLegacyAppRootDir, legacyAppRootDir),
                newHostPathVolume(volumeNameSysDir, volumePathSysDir),
                newHostPathVolume(volumeNameDevDir, volumePathDevDir),
                newHostPathVolume(volumeNameRunUdevData, volumePathRunUdevData),
@@ -112,6 +115,7 @@ func getVolumesAndMounts(pluginSocketDir, certsSecretName string) (volumes []cor
                newVolumeMount(volumeNameMountpointDir, kubeletDirPath+"/pods", corev1.MountPropagationBidirectional, false),
                newVolumeMount(volumeNamePluginDir, kubeletDirPath+"/plugins", corev1.MountPropagationBidirectional, false),
                newVolumeMount(volumeNameAppRootDir, appRootDir, corev1.MountPropagationBidirectional, false),
+               newVolumeMount(volumeNameLegacyAppRootDir, legacyAppRootDir, corev1.MountPropagationBidirectional, false),
                newVolumeMount(volumeNameSysDir, volumePathSysDir, corev1.MountPropagationBidirectional, true),
                newVolumeMount(volumeNameDevDir, volumePathDevDir, corev1.MountPropagationHostToContainer, true),
                newVolumeMount(volumeNameRunUdevData, volumePathRunUdevData, corev1.MountPropagationBidirectional, true),

```